test/01_typedef.jl

```diff
@@ -39,7 +39,7 @@ module TestTypeDef
     for i in 1:3
         x = CategoricalValue(i, pool)
 
-        @test isa(x, CategoricalValue)
+        @test iscatvalue(x)
```

Comment:
I agree that there should be more tests covering different categorical types.
But how much should there be emphasis on the actual implementation?
I suppose as long as the "black box" is doing the right thing (e.g. `AbstractString` methods are working as expected on string categories) it doesn't matter what actual types it uses.
That should reduce the burden of tests maintenance.

Reply:
Since 80% of methods are common to all `CatValue` types, we don't really have to replicate the test for `CategoricalValue` and `CategoricalString`. Indeed as long as the "black box" works it should be OK, but AFAIK we only test the `AbstractString` interface on `CategoricalString` separately, and not in the context of `CategoricalArray{String}` (or at least not systematically). So it would be good to test in a few situations that we indeed get a `CategoricalString` or a `CategoricalValue` where we expect it (i.e. when indexing an array constructed by a few different means). Luckily these tests already exist, they just need an additional line where appropriate.